### PR TITLE
S3 | Join a Hunt

### PR DESCRIPTION
**Pull request by anthonyshibitov**
Created /:id PUT path which inserts an email specified in the body of the request into the Mongo collection 'users' to the respective id specified in the params.

Created a /list GET path which returns the contents of the 'hunts' collection, for debugging purposes.

#49

### CODE REVIEW

**Comment by anthonyshibitov** [comment]
Also updated .gitignore file

File: hunts-api/routes/hunts.js [hunk 1]
```diff
@@ -2,15 +2,49 @@
 
 const express = require('express');
 const router = express.Router();
+const getDb = require('../db/index.js');
 
+// TODO: remove when deployed to production, this is for debugging only !!
+router.get('/list', (req, res) => {
+  process.stdout.write("Connecting to Mongo ...\n");
```
Comment (raghunat) [inline hunts-api/routes/hunts.js]: not a problem, but this is the same as `console.log`

File: hunts-api/routes/hunts.js [hunk 2]
```diff
@@ -2,15 +2,49 @@
...
+  getDb().then(db => {
+    process.stdout.write("Connected\n");
+    db.collection('hunts').find(req.params.id).toArray((err, docs) => {
+  		if(err){
+  			throw err;
```
Comment (raghunat) [inline hunts-api/routes/hunts.js]: instead of throw err, lets ` return res.status(500).json(err)`

File: hunts-api/routes/hunts.js [hunk 3]
```diff
@@ -2,15 +2,49 @@
...
+  	});
+  }).catch(e => res.status(500).json(e));
+});
 
-router.post('/', (req,res) => {
-    res.json({
-        name: 'Hunt 1'
+router.put('/:id', (req, res) => {
+  if(!req.body.email || !req.params.id){
+    res.status(403).json({message: "Invalid information."});
```
Comment (raghunat) [inline hunts-api/routes/hunts.js]: You need to add return here, otherwise it will keep going.

File: hunts-api/routes/hunts.js [hunk 4]
```diff
@@ -2,15 +2,49 @@
...
+    db.collection('hunts')
+    .findOneAndUpdate({id: req.params.id},{
+      $push: {
+        email: req.body.email
+      }
+    }, {
+      upsert: true
+    }, (err, result) => {
+      if(err){
+        throw err;
```
Comment (raghunat) [inline hunts-api/routes/hunts.js]: Same as above

File: hunts-api/routes/hunts.js [hunk 5]
```diff
@@ -2,15 +2,49 @@
...
+  if(!req.body.email || !req.params.id){
+    res.status(403).json({message: "Invalid information."});
+  }
+  process.stdout.write("Connecting to Mongo ...\n");
+  getDb().then(db => {
+    process.stdout.write("Connected to Mongo\n");
+    db.collection('hunts')
+    .findOneAndUpdate({id: req.params.id},{
+      $push: {
+        email: req.body.email
```
Comment (raghunat) [inline hunts-api/routes/hunts.js]: The field to push into is `currentUsers `

**Pull request merged**